pkg/controllers/disruption/consolidation.go

```diff
@@ -198,6 +204,76 @@ func (c *consolidation) computeConsolidation(ctx context.Context, candidates ...
 	}, nil
 }
 
+// Compute command to execute spot-to-spot consolidation if:
+//  1. The SpotToSpotConsolidation feature flag is set to true.
+//  2. For single-node consolidation:
+//     a. There are at least 15 cheapest instance type replacement options to consolidate.
+//     b. The current candidate is NOT part of the first 15 cheapest instance types inorder to avoid repeated consolidation.
+func (c *consolidation) computeSpotToSpotConsolidation(ctx context.Context, candidates []*Candidate, results *pscheduling.Results,
+	candidatePrice float64) (Command, error) {
+
+	// Spot consolidation is turned off.
+	if !options.FromContext(ctx).FeatureGates.SpotToSpotConsolidation {
+		if len(candidates) == 1 {
+			c.recorder.Publish(disruptionevents.Unconsolidatable(candidates[0].Node, candidates[0].NodeClaim, "SpotToSpotConsolidation is disabled, can't replace a spot node with a spot node")...)
+		}
+		return Command{}, nil
+	}
+
+	// Since we are sure that the replacement nodeclaim considered for the spot candidates are spot, we will enforce it through the requirements.
+	results.NewNodeClaims[0].Requirements.Add(scheduling.NewRequirement(v1beta1.CapacityTypeLabelKey, v1.NodeSelectorOpIn, v1beta1.CapacityTypeSpot))
+	// All possible replacements for the current candidate compatible with spot offerings
+	instanceTypeOptionsWithSpotOfferings :=
+		results.NewNodeClaims[0].NodeClaimTemplate.InstanceTypeOptions.Compatible(results.NewNodeClaims[0].Requirements)
+
+	// Possible replacements that are lower priced than the current candidate
+	results.NewNodeClaims[0].NodeClaimTemplate.InstanceTypeOptions = filterByPrice(instanceTypeOptionsWithSpotOfferings, results.NewNodeClaims[0].Requirements, candidatePrice)
+
+	if len(results.NewNodeClaims[0].NodeClaimTemplate.InstanceTypeOptions) == 0 {
+		if len(candidates) == 1 {
+			c.recorder.Publish(disruptionevents.Unconsolidatable(candidates[0].Node, candidates[0].NodeClaim, "Can't replace spot node with a cheaper spot node")...)
+		}
+		logging.FromContext(ctx).With("nodeClaim", candidates[0].Name()).Debugf("can't replace spot node with a cheaper spot node.")
+		// no instance types remain after filtering by price
+		return Command{}, nil
+	}
+
+	// For multi-node consolidation:
+	if len(candidates) > 1 {
+		return Command{
+			candidates:   candidates,
+			replacements: results.NewNodeClaims,
+		}, nil
+	}
+
+	// For single-node consolidation:
+
+	// Allow only if there are minimum number of possible InstanceTypes for a NodeClaim
```

Comment:
Do we need to do both of these checks? Can we just collapse this into the concept of, "Couldn't get 15 cheaper instance options then current candidate"

Reply:
From offline discussion, merged these 2 conditions into one.